xrpl/models/transactions/oracle_set.py

```diff
@@ -0,0 +1,48 @@
+"""Model for OracleSet transaction type."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from xrpl.models.nested_model import NestedModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class OracleSet(Transaction):
+    """Represents a OracleSet transaction."""
+
+    account: str = REQUIRED  # type: ignore
+    oracle_document_id: int = REQUIRED  # type: ignore
+
+    """
+    The below three fields must be hex-encoded. You can
+    use `xrpl.utils.str_to_hex` to convert a UTF-8 string to hex.
+    """
+    provider: Optional[str] = None
+    uri: Optional[str] = None
+    asset_class: Optional[str] = None
+
+    last_update_time: int = REQUIRED  # type: ignore
+    price_data_series: List[PriceData] = REQUIRED  # type: ignore
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.ORACLE_SET,
+        init=False,
+    )
+
```

Comment:
There should be some validations on this model, like checking the length of the price data series and the last update time.

Reply:
fixed in [9c7ce5e](https://github.com/XRPLF/xrpl-py/pull/701/commits/9c7ce5e9e7cabdb5105c193447a2deb27bc6f59e)

I didn't add validations for `LastUpdateTime` because I'll need access to the Last closed time of the ledger and the existing `LastUpdateTime` of the LedgerEntry -- to perform the said validations

Reply:
You can do some of the checks, like ensuring it's a valid time.
To refer to the C++ code:
```cpp
if (lastUpdateTime < epoch_offset.count())
        return tecINVALID_UPDATE_TIME;
```

Reply:
okay, I've upadted in 49370f1

I don't think this check is useful, because old time-points fail the 300s difference check -- with respect to the LastClosedLedger.

It could have been a `tem...` error to avoid the fees of the mistake.